internal/database/gitserver_repos.go

```diff
@@ -495,6 +522,44 @@ WHERE
 	return nil
 }
 
+func (s *gitserverRepoStore) LogCorruption(ctx context.Context, name api.RepoName, reason string) error {
+	// trim reason to 1 MB so that we don't store huge reasons and run into trouble when it gets too large
+	if len(reason) > MaxReasonSizeInMB {
+		reason = reason[:MaxReasonSizeInMB]
+	}
+
+	log := types.RepoCorruptionLog{
+		Timestamp: time.Now(),
+		Reason:    reason,
+	}
+	var rawLog []byte
+	if data, err := json.Marshal(log); err != nil {
+		return errors.Wrap(err, "could not marshal corruption_logs")
+	} else {
+		rawLog = data
+	}
+
+	res, err := s.ExecResult(ctx, sqlf.Sprintf(`
+UPDATE gitserver_repos as gtr
+SET
+	corrupted_at = NOW(),
+	-- append the json and then ensure we only keep 10 items in the resulting json array
+	corruption_logs = (SELECT jsonb_path_query_array(%s||gtr.corruption_logs, '$[0 to 9]')),
+	updated_at = NOW()
+WHERE repo_id = (SELECT id FROM repo WHERE name = %s) AND corrupted_at IS NULL
```

Comment:
Heya, it's ya pal, Mr Nit Pick

```suggestion
WHERE
      repo_id = (SELECT id FROM repo WHERE name = %s)
AND
      corrupted_at IS NULL
```

(These need to be tabs, which GitHub won't let me type)

Reply:
fixed!